components/sup/src/manager/service/mod.rs

```diff
@@ -935,14 +889,17 @@ impl Service {
             }
         } else {
             self.check_process();
-            // NOTE: if you need reconfiguration and you DON'T have a
-            // reload script, you're going to restart anyway.
-            if self.needs_reload || self.process_down() || self.needs_reconfiguration {
-                self.reload(launcher);
-                if self.needs_reconfiguration {
-                    // NOTE this only runs the hook if it's defined
-                    self.reconfigure()
-                }
+            if hook_update_table.run
+               || hook_update_table.post_run
+               || self.process_down()
+               || (self.hooks.reconfigure.is_none() && config_change)
+            {
+                // TODO (DM): This flag is a hack. We have the `TaskExecutor` here. We could just
+                // schedule the `stop` future, but the `Manager` wraps the `stop` future with
+                // additional functionality. Can we refactor to make this flag unnecessary?
```

Comment:
It is not ideal, but I am going to propose we leave this TODO for the time being. I am hopeful that as we continue refactoring the lifecycle hooks a better pattern will emerge. For example, if we end up with a `Service` future we could just spawn the future directly even without the `TaskExecutor`.

Reply:
I'm cool with letting this sit for a while and seeing how things evolve. Using a return value was an alternative that sprang to my mind, but given the state of other code surrounding this, that might make things more complicated.

It would be good to add some comments on this field stating what it's currently intended to do, and that we're ultimately looking to remove it, just so Future Us doesn't lose sight of this.